operator/v1/types_dns.go

```diff
@@ -0,0 +1,102 @@
+// for pods and services in the cluster.
+//
+// This supports DNS based service discovery specification:
+// https://github.com/kubernetes/dns/blob/master/docs/specification.md
+//
+// More details: https://kubernetes.io/docs/tasks/administer-cluster/coredns
+type DNS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the DNS.
+	Spec DNSSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the DNS.
+	Status DNSStatus `json:"status,omitempty"`
+}
+
+// DNSSpec is the specification of the desired behavior of the DNS.
+type DNSSpec struct {
+}
+
+// DNSStatus defines the observed status of the DNS.
+type DNSStatus struct {
+	// ClusterIP is the service IP through which this DNS is made available.
+	//
+	// In the case of the default DNS, this will be a well known IP that is used
+	// as the default nameserver for pods that are using the default ClusterFirst DNS policy.
+	//
+	// In general, this IP can be specified in a pod's spec.dnsConfig.nameservers list
+	// or used explicitly when performing name resolution from within the cluster.
+	// Example: dig foo.com @<service IP>
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	ClusterIP string `json:"clusterIP"`
+
+	// ClusterDomain is the local cluster DNS domain suffix for DNS services.
+	// This will be a domain name as defined in RFC 1034 (A-Z, a-z, 0-9 and the hyphen)
+	// Example: "cluster.local"
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service
+	ClusterDomain string `json:"clusterDomain"`
+
+	// Conditions represent the latest available observations of a
+	// DNS's current state.
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []DNSCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// DNSCondition contains details for the current condition of this DNS.
+type DNSCondition struct {
+	// Type is the type of the condition.
+	Type DNSConditionType `json:"type"`
+
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the last time a condition status transitioned from
+	// one state to another.
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition.
+	Reason string `json:"reason,omitempty"`
+
+	// Message is a human readable description of the details of the last
+	// transition, complementing reason.
+	Message string `json:"message,omitempty"`
+}
+
+// DNSConditionType is a valid value for DNSCondition
+type DNSConditionType string
+
+const (
+	// DNSAvailable means the DNS is able to service requests.
+	DNSAvailable DNSConditionType = "Available"
+	// TODO: add other DNS condition types
```

Comment:
Is this a TODO for a future PR, or something you intended to add before this PR merged?